models/DataObject/ClassDefinition/Data/Block.php

```diff
@@ -30,7 +30,7 @@
 use Pimcore\Normalizer\NormalizerInterface;
 use Pimcore\Tool\Serialize;
 
-class Block extends Data implements CustomResourcePersistingInterface, ResourcePersistenceAwareInterface, LazyLoadingSupportInterface, TypeDeclarationSupportInterface, VarExporterInterface, NormalizerInterface, DataContainerAwareInterface, PreGetDataInterface, PreSetDataInterface, FieldDefinitionEnrichmentModelInterface
+class Block extends Data implements CustomResourcePersistingInterface, ResourcePersistenceAwareInterface, LazyLoadingSupportInterface, TypeDeclarationSupportInterface, VarExporterInterface, NormalizerInterface, DataContainerAwareInterface, PreGetDataInterface, PreSetDataInterface, FieldDefinitionEnrichmentModelInterface, ClassSavedInterface
```

Comment:
This is wrong, because it have the DataContainerAwareInterface

Reply:
@blankse I am not sure why this should be a problem, because the `DataContainerAwareInterface` doesn't include the `classSaved` method

Reply:
@robertSt7 See the if before your change
```
if (!$fd instanceof ClassDefinition\Data\DataContainerAwareInterface && method_exists($fd, 'classSaved')) {
```
It should be only called if it has not the DataContainerAwareInterface